src/Datadog.Trace.ClrProfiler.Managed/CallTarget/CallTargetState.cs

```diff
@@ -7,22 +7,52 @@ namespace Datadog.Trace.ClrProfiler.CallTarget
     /// </summary>
     public readonly struct CallTargetState
     {
+        private readonly Scope _oldScope;
+        private readonly Scope _scope;
         private readonly object _state;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="CallTargetState"/> struct.
         /// </summary>
+        /// <param name="scope">Scope instance</param>
+        public CallTargetState(Scope scope)
+        {
+            _oldScope = null;
+            _scope = scope;
+            _state = null;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CallTargetState"/> struct.
+        /// </summary>
+        /// <param name="scope">Scope instance</param>
         /// <param name="state">Object state instance</param>
-        public CallTargetState(object state)
+        public CallTargetState(Scope scope, object state)
         {
+            _oldScope = null;
+            _scope = scope;
             _state = state;
         }
 
+        private CallTargetState(Scope oldScope, Scope scope, object state)
+        {
+            _oldScope = oldScope;
```

Comment:
[Non-blocking] Can we rename "oldScope" to "previousScope" everywhere? You did it below in `WithPreviousScope()` 😅

Reply:
You're right `previousScope` is a better name, changed in 1632d06